libraries/lib-module-manager/PluginManager.h

```diff
@@ -174,9 +37,45 @@ typedef wxArrayString PluginIDs;
 
 class PluginRegistrationDialog;
 
+class MODULE_MANAGER_API PluginValidator
+{
+public:
+   class MODULE_MANAGER_API Callback
+   {
+   public:
+      virtual ~Callback();
+      virtual void OnValidatorError(const wxString& error) = 0;
+      virtual void OnPluginValidationResult(const wxString& providerId, const wxString& pluginPath, bool valid, const wxString& error) = 0;
+   };
+   
+   virtual ~PluginValidator();
+
+   virtual void Validate(const wxString& providerId, const wxString& path) = 0;
+
+   virtual void Initialize(Callback* callback) = 0;
+   virtual void Finalize() = 0;
+};
+
 class MODULE_MANAGER_API PluginManager final : public PluginManagerInterface
 {
 public:
+   class MODULE_MANAGER_API PluginCheckDelegate
+   {
+   public:
+      virtual ~PluginCheckDelegate();
+
+      virtual void OnPluginValidationStarted(const wxString& path, int index, int count) = 0;
+      virtual void OnPluginValidationFinished() = 0;
+      virtual void OnPluginValidationResult(const wxString& path, bool valid) = 0;
```

Comment:
I don't believe in using `noexcept` too widely, but functions meant to be called inside a `finally` and therefore inside a destructor should have it.

Reply:
That applies to `OnPluginValidationFinished `